Studies/python/CalculateWeigths.py

```diff
@@ -0,0 +1,197 @@
+
+    for sample_type in range(4):
+        for spin in range(2):
+            for mass in range(xs_all.shape[2]):
+                weights[:,:,sample_type,spin,mass] *= xs_all[sample_type, spin, mass]
+
+    # Radion = Graviton
+    indices = mass_node_unique >= 250
+    res_process = [ggHH_Res, VBFHH_Res]
+    non_res_process = [ggHH_NonRes, VBFHH_NonRes]
+    all_process = [res_process, non_res_process]
+    for year in range(3):
+        for channel in range(3):
+            for prod_mode in res_process:
+                radion = np.sum(weights[year, channel, prod_mode, 0, :] * counts[year, channel, prod_mode, 0, :])
+                graviton = np.sum(weights[year, channel, prod_mode, 1, :] * counts[year, channel, prod_mode, 1, :])
+                if radion == 0 or graviton == 0: continue
+                C = radion / graviton
+                weights[year, channel, prod_mode, 1, :] *= C
+
+    # VBF = ggF
+    for year in range(3):
+        for channel in range(3):
+            for x in all_process:
+                VBF = np.sum(weights[year, channel, x[1], :, :] * counts[year, channel, x[1], :, :])
+                ggF = np.sum(weights[year, channel,  x[0], :, :] * counts[year, channel,  x[0], :, :])
+                if VBF == 0 or ggF == 0: continue
+                E = VBF / ggF
+                weights[year, channel, x[0], :, :] *= E
+
+    # Res = Non Res
+    for year in range(3):
+        for channel in range(3):
+            non_res = np.sum(weights[year, channel, non_res_process, :, :] * counts[year, channel, non_res_process, :, :])
+            res = np.sum(weights[year, channel, res_process, :, :] * counts[year, channel, res_process, :, :])
+            if non_res == 0 or res == 0: continue
+            D = non_res / res
+            weights[year, channel, res_process, :, :] *= D
+
+
+    # eTau = muTau = tauTau
+    for year in range(3):
+        ch_cnt = np.zeros(3)
+        for channel in range(3):
+            ch_cnt[channel] = np.sum(weights[year, channel, :, :, :] * counts[year, channel, :, :, :])
+
+        ref_ch = np.amax(ch_cnt)
+        if ref_ch == 0: continue
+
+        for channel in range(3):
+            if ch_cnt[channel] != 0 :
+                A = ref_ch / ch_cnt[channel]
+                weights[year, channel, :, :, :] *= A
+
+    # all years
+    year_cnt = np.zeros(3)
+    for year in range(3):
+         year_cnt[year] = np.sum(weights[year, :, :, :, :] * counts[year, :, :, :, :])
+
+    ref_year = np.amax(year_cnt)
+    if ref_year == 0:
+        raise RuntimeError('bad year')
+
+    for year in range(3):
+        if year_cnt[year] != 0 :
+            A = ref_year / year_cnt[year]
+            weights[year, :, :, :, :] *= A
+
+    #sum all = n_evt
+    weights_count = np.sum(weights * counts)
+    C =  X.shape[0] / weights_count
+
+    weights *= C
+
+    return weights, mass_node_unique, mass_node
+
+# @numba.njit
+def ConvertToVector(X,Z):
```

Comment:
- ConvertToVector should have a numba decorator
- Create another method called `CreateSampleWeights` that will be used from outside